send.go

```diff
@@ -198,6 +202,64 @@ func (cli *Client) SendMessage(ctx context.Context, to types.JID, message *waPro
 	}
 	resp.ID = req.ID
 
+	isInlineBotMode := false
+
+	if !req.InlineBotJID.IsEmpty() {
+		if !req.InlineBotJID.IsBot() {
+			err = ErrInvalidInlineBotID
+			return
+		}
+		isInlineBotMode = true
+	}
+
+	isBotMode := isInlineBotMode || to.IsBot()
+	var botNode = waBinary.Node{
+		Tag: "bot",
+	}
+
+	if isBotMode {
+		// common code for both inline and not inline modes
+		message.MessageContextInfo = &waE2E.MessageContextInfo{
+			BotMetadata: &waE2E.BotMetadata{
+				PersonaID: &req.BotPersonaID,
+			},
+			MessageSecret: random.Bytes(32),
+		}
```

Comment:
This is what I meant by allowing custom message secrets:

```suggestion
		if message.MessageContextInfo == nil {
			message.MessageContextInfo = &waE2E.MessageContextInfo{}
		}
		if message.MessageContextInfo.MessageSecret == nil {
			message.MessageContextInfo.MessageSecret = random.Bytes(32)
		}
```

Also, since the persona ID already has a field in the protobuf, I don't think it needs a special extra param. If the persona ID is required, then it can be set if missing the same way as MessageSecret

Reply:
I agree with this (allow for custom message secret) in general but the persona ID seems to be required